### DISCUSSION TITLE
Custom Filter Function in DataTable Not Adapting to 'AND'/'OR' Filter Operators

### DISCUSSION THREAD

**Discussion by juancamiloqhz** [primefaces/community#344] [PrimeReact]
Hello all,

I am implementing PrimeReact version 9.6.1 into my project and using a custom filter function to filter certain columns within a DataTable. The filter function I'm using right now looks like this:

```javascript
const customFunction = (value, filter) => {
    if (filter !== null && typeof value === 'string') {
        const valueArr = value.split(', ');
        const filtersArr = Array.isArray(filter) ? filter : filter.split(', ');
        return filtersArr.every((r) => valueArr.includes(r.trim()));
    }
    return true;  // If filter is null, all records would pass the filter.
};
```

The issue I'm encountering emerges when I alter the filter operator. As per my current implementation, the filter function is set to evaluate every value. But what I need is for the customFunction to adjust based on whether the filter operator is 'AND' ('every') or 'OR' ('some').

At the present moment, whenever I switch between the filter operators, the function repeats the same operation irrespective of the operator's value. My objective is to have the filtering function adapt to filter by 'AND' or 'OR' accordingly when the filter operator is altered.

**Comment by gucal** [comment]
Hi,

If you think the CustomFilter prop is not working as it should be, you can create an github issue. We can work on it. 

Thank you.

**Reply by juancamiloqhz** [reply]
Thank you for your swift reply, @gucal.

My goal is to identify the active filter operator in the column where I'm implementing a custom filter function. Here's a hypothetical implementation:

```javascript
const customFunction = (value, filter, filterLocale, params) => {
    if (!filter || typeof value !== 'string') return true;

    const filterOperator = params.filterOperator;
    const valueArr = value.split(', ');
    const filtersArr = Array.isArray(filter) ? filter : filter.split(', ');

    if (filterOperator === FilterOperator.AND) {
        return filtersArr.every((r) => valueArr.includes(r.trim()));
    } else if (filterOperator === FilterOperator.OR) {
        return filtersArr.some((r) => valueArr.includes(r.trim()));
    }
    
    // If no operator matches, return true so all records pass the filter.
    return true;
};
```

However, I'm struggling to find the filter operator within the params argument or any other arguments in the custom filter function. I am presenting this issue to you in case there exists a `filterOperator` parameter that I may be unacquainted with, or it might be located somewhere else.

**Reply by melloware** [reply]
Does this Custom Filter function example help: https://primereact.org/datatable/#custom_filter

**Comment by mertsincan** [comment]
@gucal ?
